hashids-with-django-secret

Sjord opened the pull request:
The Django secret key is used as salt in HashIDs. The HashID mechanism is not secure. By observing sufficient HashIDs, the salt used to construct them can be recovered. This means the Django secret key can be obtained by attackers, through the HashIDs.

This is a rare bug, but easy to detect.

Sjord committed:
hashids-with-django-secret

The Django secret key is used as salt in HashIDs. The HashID mechanism is not secure.
By observing sufficient HashIDs, the salt used to construct them can be recovered.
This means the Django secret key can be obtained by attackers, through the HashIDs.

This is a rare bug, but easy to detect.

Sjord committed:
hashids-with-flask-secret

The Flask secret key is used as salt in HashIDs. The HashID mechanism is not secure.
By observing sufficient HashIDs, the salt used to construct them can be recovered.
This means the Flask secret key can be obtained by attackers, through the HashIDs.

0xDC0DE committed:
Merge branch 'develop' into hashids-with-django-secret

0xDC0DE commented on python/django/security/hashids-with-django-secret.yaml:
```diff
@@ -0,0 +1,26 @@
+    By observing sufficient HashIDs, the salt used to construct them can be recovered.
+    This means the Django secret key can be obtained by attackers, through the HashIDs.
+  metadata:
+    category: security
```
Looks like you are missing a subcategory

https://semgrep.dev/docs/contributing/contributing-to-semgrep-rules-repository/#subcategory

0xDC0DE commented on python/flask/security/hashids-with-flask-secret.yaml:
```diff
@@ -0,0 +1,33 @@
+    By observing sufficient HashIDs, the salt used to construct them can be recovered.
+    This means the Flask secret key can be obtained by attackers, through the HashIDs.
+  metadata:
+    category: security
```
and here

Sjord committed:
Add subcategory vuln

Linter complained about subcategory missing.
This seems like an actual problem that should be resolved.

0xDC0DE committed:
Merge branch 'develop' into hashids-with-django-secret

0xDC0DE committed:
Merge branch 'develop' into hashids-with-django-secret
